File: app/src/main/java/top/easelink/lcg/ui/main/articles/viewmodel/ArticleEmptyItemViewModel.java
package top.easelink.lcg.ui.main.articles.viewmodel;

public class ArticleEmptyItemViewModel {

    private ArticleEmptyItemViewModelListener mListener;

    public ArticleEmptyItemViewModel(ArticleEmptyItemViewModelListener listener) {
        this.mListener = listener;
    }

    public void onRetryClick() {
        mListener.onRetryClick();
    }

    public interface ArticleEmptyItemViewModelListener {

        void onRetryClick();
    }
}

File: app/src/main/java/top/easelink/lcg/ui/main/articles/viewmodel/ArticleEmptyItemViewModel.kt
package top.easelink.lcg.ui.main.articles.viewmodel

class ArticleEmptyItemViewModel internal constructor(private val mListener: ArticleEmptyItemViewModelListener) {
    fun onRetryClick() {
        mListener.onRetryClick()
    }

    interface ArticleEmptyItemViewModelListener {
        fun onRetryClick()
    }

}